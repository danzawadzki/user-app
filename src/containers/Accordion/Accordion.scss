@import '../../assets/styles/colors';

.Accordion {
	&__heading {
		color: $brand--accent;
		text-decoration: underline;
		cursor: pointer;

		&:hover {
			color: lighten($brand--accent, 10);
		}
	}

	&__body {
		transition: height ease-out 1s;
		overflow: auto;

		&--visible {
		}
		&--hidden {
			transition: height ease-out 1s;
			height: 0;
		}
	}
}
