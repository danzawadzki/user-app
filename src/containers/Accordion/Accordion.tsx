import * as React from 'react';
import { Component } from 'react';
import './Accordion.scss';

/** Interface for accordion props */
export interface IAccordionProps {
	/** Accordion heading */
	heading: {
		/** To display, when accordion is open */
		on: string;
		/** To display, when accordion is closed */
		off: string;
	};
	/** Accordion body content */
	children: JSX.Element | JSX.Element[] | string;
}

/** Interface for accordion state */
export interface IAccordionState {
	/** Flag to open and close accordions */
	isOpen: boolean;
}

/**
 * Accordion component.
 *
 * @version 1.0.0
 */
export default class Accordion extends Component<IAccordionProps, IAccordionState> {
	/**
	 * Function to handle visibility flag toggling
	 */
	toggleIsOpenFlag = () => {
		this.setState(prevState => ({
			isOpen: !prevState.isOpen
		}));
	};

	constructor(props: IAccordionProps) {
		super(props);

		this.state = {
			isOpen: true
		};
	}

	render() {
		const { heading, children } = this.props;
		const { isOpen } = this.state;

		return (
			<div className="Accordion">
				<a onClick={this.toggleIsOpenFlag} className="Accordion__heading">
					{isOpen ? heading.on : heading.off}
				</a>
				<div
					className={`Accordion__body Accordion__body--${
						isOpen ? 'visible' : 'hidden'
					}`}>
					{children}
				</div>
			</div>
		);
	}
}
