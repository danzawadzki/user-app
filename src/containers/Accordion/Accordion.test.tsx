import React from 'react';
import { shallow } from 'enzyme';
import Accordion from './Accordion';

describe('UserCard component: ', () => {
	//Shallow rendered component
	const heading = { on: 'Hide', off: 'Show' };
	const body = 'Accordion test body content.';

	const component = shallow(<Accordion heading={heading}>{body}</Accordion>);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should contain on label and content', () => {
		expect(component.html()).toContain(heading.on);
		expect(component.html()).toContain(body);
	});

	it('should contain off label and not contain content after click', () => {
		component.find('a').simulate('click');
		expect(component.html()).toContain(heading.off);
		expect(component.find('.Accordion__body--hidden').exists());
	});
});
