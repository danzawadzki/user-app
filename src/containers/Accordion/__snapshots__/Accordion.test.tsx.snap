// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component:  should render match snapshot 1`] = `
<div
  className="Accordion"
>
  <a
    className="Accordion__heading"
    onClick={[Function]}
  >
    Hide
  </a>
  <div
    className="Accordion__body Accordion__body--visible"
  >
    Accordion test body content.
  </div>
</div>
`;
